### PR TITLE
PINT-387 Add Fast button widgets

### PR DESCRIPTION
**Pull request by corywebb**
# Description

Add WordPress widgets to display the Fast Checkout buttons and the Fast Login button.

# Testing instructions

1. Login to the [WordPress admin on the staging site](https://fasttestdev.wpcomstaging.com/wp-admin/).
2. Go to the [Widgets manager](https://fasttestdev.wpcomstaging.com/wp-admin/widgets.php).
3. Add the Fast Cart Button, Fast Product Button, and Fast Login Button widgets to the "Footer" position.
4. Verify that the widgets are displayed in the footer when appropriate.

# Checks
- [x] Code has been formatted with `phpcbf --standard WordPress`
- [x] No new linter warnings from `phpcs --standard WordPress`

@brikr @ilkerulutas this is ready for review.

### CODE REVIEW

File: includes/widgets/class-cart.php
Comment (brikr) [inline includes/widgets/class-cart.php:41]: what does this backslash do?

Comment (corywebb) [reply includes/widgets/class-cart.php:41]: The `\` is a namespace separator. Since this file has a namespace of `FastWC\Widgets`, you have to set the root namespace on certain things to indicate that they do not belong to the current file's namespace. Putting the `\` at the beginning of a function shows that it is in the root namespace.

**Review by brikr — DISMISSED** [review]

**Comment by corywebb** [comment]
@brikr this has been updated to include Variation ID and Quantity fields in the widget admin. You can see this functionality working at the bottom of the [staging home page](https://fasttestdev.wpcomstaging.com/) with a button to order 1 large beanie. Can you review this one more time before I merge it?

I intend to add product options later, but I need to think through how to accomplish that with the widget UI.

File: templates/buttons/fast-checkout-button.php
Comment (brikr) [inline templates/buttons/fast-checkout-button.php:24]: there is also a `product_options` attribute supported, it's a JSON map of option names to their values. You can see it in use on this page: https://fasttestdev.wpcomstaging.com/2021/06/24/brik-testing-headless-checkout/

**Review by brikr — APPROVED** [review]
LGTM as is, but if you want to add the product_options support now that's great too

File: templates/buttons/fast-checkout-button.php
Comment (corywebb) [reply templates/buttons/fast-checkout-button.php:24]: Thanks @brikr. I'm going to hold off on the `product_options` attribute while I think about the best way to implement the UI for adding the options.

**Pull request merged**
